### DISCUSSION TITLE
Check if form contains defaultValues

### DISCUSSION THREAD

**Discussion by FilipBrannlund** [react-hook-form/react-hook-form#5437] [Q&A] [answered]
I want to know if the user is trying to submit a form that contains the default values.
dirtyFields/isDirty does not help me becaus the user can manipulate inputfields = form is now dirty but then change back the input to its default value = form is still dirty but I want to know if inputs in the form has values that is not equal to default value.

In CSB below, 
checking an checkbox
Form is now dirty
unckecking the checkbox 
Form is still dirty but the form values is as default
Here I want to see if the values in form is not the same as the defaultvalues.

https://codesandbox.io/s/custom-hook-with-resolver-v7-forked-04noe?file=/src/App.js

**Comment by bluebill1049** [comment]
> form is now dirty but then change back the input to its default value = form is still dirty but I want to know if inputs in the form has values that is not equal to default value.

If user changed form value it will be dirty, if they change it back to default value it will not be dirty anymore.

**Comment by thanh-nguyen-95** [comment, marked as answer]
I think the default value of each `car` are changed when registered. At first they are `undefined` but could be `false` after the first render.

Setting default value for each `car` should fix the problem:

```jsx
const defaultValues = {
  unsatisfied: false,
  cars: Array(1000)
    .fill(1)
    .map((_, index) => ({ type: `Car ${index}`, color: false, engine: false, wheels: false }))
};
```

Working CodeSandbox: https://codesandbox.io/s/custom-hook-with-resolver-v7-forked-lyurb

'Send' button will be disabled if form is not dirty.

**Reply by bluebill1049** [reply]
thank you. @thanh-nguyen-95
